File: MKAgent/MsgType.java
package MKAgent;
/**
 * Types of messages the game engine can send to the agent.
 */
public enum MsgType
{
	/**
	 * message announcing the start of the game ("new_match" message)
	 */
	START,
	/**
	 * message describing a move or a swap ("state_change" message)
	 */
	STATE,
	/**
	 * message informing about the end of the game ("game_over" message)
	 */
	END
}

File: magent/protocol/msg_type.py
from enum import Enum


class MsgType(Enum):
    # message announcing the start of the game ("new_match" message)
    START = 'new_match'
    # message describing a move or a swap ("state_change" message)
    STATE = 'state_change'
    # message informing about the end of the game ("game_over" message)
    END = 'game_over'

